src/encoding/armor.js

```diff
@@ -247,6 +247,9 @@ function splitChecksum(text) {
   if (lastEquals >= 0) {
     body = text.slice(0, lastEquals);
     checksum = text.slice(lastEquals + 1);
+    if (checksum.substr(0, 6) === '\n-----') {
```

Comment:
This part confuses me. I think you are trying to handle the case when the base64-encoded data itself ends in = but the checksum is missing right? I don't think this condition works, because the ----- headers are already stripped off at this point so it will never match. A better approach I think would be to trim the input string to remove trailing whitespace and then test for lastEquals === text.length - 1. If that is true, that = is part of the body and you need to set body = text and checksum = ''. Does that make sense or am I missing something?